lib/internal/Magento/Framework/App/Test/Unit/HttpTest.php

```diff
@@ -174,13 +175,72 @@ public function testLaunchException()
         $this->frontControllerMock->expects($this->once())->method('dispatch')->with($this->requestMock)->will(
             $this->returnCallback(
                 function () {
+                    // phpcs:ignore Magento2.Exceptions.DirectThrow
                     throw new \Exception('Message');
                 }
             )
         );
         $this->http->launch();
     }
 
+    /**
+     * Test that HEAD requests lead to an empty body and a Content-Length header matching the original body size.
+     * @dataProvider dataProviderForTestLaunchHeadRequest
+     * @param string $body
+     * @param int $expectedLength
+     */
+    public function testLaunchHeadRequest($body, $expectedLength)
+    {
+        $this->setUpLaunch();
+        $this->requestMock->expects($this->once())->method('isHead')->will($this->returnValue(true));
+        $this->responseMock->expects($this->once())
+            ->method('getHttpResponseCode')
+            ->will($this->returnValue(200));
+        $this->responseMock->expects($this->once())
+            ->method('getContent')
+            ->will($this->returnValue($body));
+        $this->responseMock->expects($this->once())
+            ->method('clearBody')
+            ->will($this->returnValue($this->responseMock));
+        $this->responseMock->expects($this->once())
+            ->method('setHeader')
+            ->with('Content-Length', $expectedLength)
+            ->will($this->returnValue($this->responseMock));
+        $this->eventManagerMock->expects($this->once())
+            ->method('dispatch')
+            ->with(
+                'controller_front_send_response_before',
+                ['request' => $this->requestMock, 'response' => $this->responseMock]
+            );
+        $this->assertSame($this->responseMock, $this->http->launch());
+    }
+
+    /**
+     * Different test content for responseMock with their expected lengths in bytes.
+     * @return array
+     */
+    public function dataProviderForTestLaunchHeadRequest()
```

Comment:
The return type is missed.

Reply:
Could you elaborate, please?

Reply:
You specified the doc block return type but missed the method return type `dataProviderForTestLaunchHeadRequest(): array`.